sympy/physics/quantum/tests/test_qexpr.py

```diff
@@ -31,6 +31,7 @@ def test_qexpr_commutative():
     q = QExpr._new_rawargs(0, 1, HilbertSpace())
     assert q.is_commutative is False
 
+@XFAIL
```

Comment:
`QExpr` does not currently play well with sympy's more basic components.  Sometimes it wants to act like a single symbol representing a state or an operator, but it's composed of several indices or coordinate variables.  So, in `master`,   `QExpr.free_symbols` returns a set containing itself, instead of the index parameters it is composed of which are themselves potentially free for assignment.

The problem then comes when we need to hash an expression containing a quantum state or operator subclassing `QExpr`.  In `sympy/physics/quantum/tests/test_spin.py` there are several
sums over index variables appearing inside quantum states or operators.  To hash these, we have to hash a `QExpr`.  Since a `QExpr` is pretending to be a raw symbol, the current `__hash__` routines treat it as such and attempt to canonize it, which requires making use of `QExpr.name`, which doesn't actually exist, so raises an error.

Possible solutions:
1) On solution might be to add the right `QExpr.name`, but I'm unsure what that would be. 
2) Change `QExpr` so that instead of appearing atomic, it's index variables are visible through
`QExpr.free_symbol`.  That seems more natural to me, so it's what I implemented.  But now this test fails because I've redefined the nature of `free_symbol`.
3) Change `Basic` to treat `QExpr` as a special case.
4) Overload hashing somehow in `QExpr`